tests/plenary/async/async_spec.lua

```diff
@@ -0,0 +1,50 @@
+require('plenary.async').tests.add_to_env()
+
+describe('async', function()
+  a.it('void functions can call wrapped functions', function()
+    local stat = 0
+    local saved_arg
+
+    local wrapped = a.wrap(function(inc, callback)
+      stat = stat + inc
+      callback()
+    end, 2)
+
+    local voided = a.void(function(arg)
+      wrapped(1)
+      wrapped(2)
+      wrapped(3)
+      stat = stat + 1
+      saved_arg = arg
+    end)
+
+    voided('hello')
+
+    assert(stat == 7)
+    assert(saved_arg == 'hello')
+  end)
+
+  a.it('void functions can call wrapped functions with ignored arguments', function()
+    local stat = 0
+    local saved_arg
+
+    local wrapped = a.wrap(function(inc, unused2, unused3, callback)
```

Comment:
can we check that the arguments are in their correct spots instead of ignoring them

Reply:
and make sure that they are actually nil and not an erroneously rotated callback